nautobot/core/ui/object_detail.py

```diff
@@ -894,6 +894,84 @@ def render_body_content(self, context):
         return result
 
 
+class StatsPanel(Panel):
+    def __init__(
+        self,
+        *,
+        filter_name,
+        related_field_names=None,
+        body_content_template_path="components/panel/stats_panel_body.html",
+        **kwargs,
+    ):
+        """Instantiate a `StatsPanel`.
+
+        Args:
+            related_field_names (dict): key/value pair of { <model_class>: <query_strings> }.
+            filter_name (str): a valid query filter append to the anchor tag for each stat button.
+            e.g. the `tenant` query parameter in the url `/circuits/circuits/?tenant=f4b48e9d-56fc-4090-afa5-dcbe69775b13`.
+        """
+
+        self.filter_name = filter_name
+        self.related_field_names = related_field_names
+        super().__init__(body_content_template_path=body_content_template_path, **kwargs)
+
+    def should_render(self, context):
+        """Always should render this panel as the permission is reinforced in python with .restrict(request.user, "view")"""
+        return True
+
+    def render_body_content(self, context):
+        """
+        Transform self.related_field_names to a dictionary with key, value pairs as follows:
+        {
+            <related_object_model_class_1>: [related_object_model_class_list_url_1, related_object_count_1, related_object_title_1],
+            <related_object_model_class_2>: [related_object_model_class_list_url_2, related_object_count_2, related_object_title_2],
+            <related_object_model_class_3>: [related_object_model_class_list_url_3, related_object_count_3, related_object_title_3],
+            ...
+        }
+        """
+        instance = context.get("obj") or context.get("object")
+        request = context["request"]
+        if isinstance(instance, TreeModel):
+            self.filter_pks = (
+                instance.descendants(include_self=True).restrict(request.user, "view").values_list("pk", flat=True)
+            )
+        else:
+            self.filter_pks = [instance.pk]
+
+        if self.body_content_template_path:
+            stats = {}
+            if not self.related_field_names:
+                return ""
+            for related_field in self.related_field_names:
+                if isinstance(related_field, tuple):
+                    related_object_model_class, query = related_field
+                else:
+                    related_object_model_class, query = related_field, f"{self.filter_name}__in"
+                filter_dict = {query: self.filter_pks}
+                related_object_count = (
+                    related_object_model_class.objects.restrict(request.user, "view").filter(**filter_dict).count()
+                )
+                related_object_model_class_meta = related_object_model_class._meta
+                related_object_list_url = get_route_for_model(related_object_model_class, "list")
```

Comment:
Possibly use `validated_viewname` here so that we don't get an error if the related model doesn't actually have a list view?